libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -525,10 +586,29 @@ void AP_Baro::update(void)
         sensors[i].healthy = (now - sensors[i].last_update_ms < 500) && !is_zero(sensors[i].pressure);
     }
 
+    // find current GPS altitude
+    float gps_alt = 0.0f;
+    bool process_gps_alt = false;
+    if (!is_zero(_gps_adj_step) && _gps_calibrated) {
+        float gps_vacc;
+        if (_gps->vertical_accuracy(gps_vacc)) {
```

Comment:
For the record when I was trying to do some of the same type of stuff (using GPS altitude to do some pressure compensation for ISA altitudes on calibration (that was way to complicated/we ditched it) I quickly ran into the same problem the I had to add it to so many places passing it in the constructor was better, but I really hated doing it, and glad I could work around it in the end. (Obviously the compensation is hard to work around)